app/crud/serviceMetric.py

```diff
@@ -1,19 +1,142 @@
-
-from sqlalchemy.orm import Session
-
-from ..models import ServiceMetric
-from ..schemas import ServiceMetricCreate, ServiceMetricUpdate
+from sqlalchemy.orm import Session 
+from sqlalchemy import func, and_
+from ..models import ServiceMetric, scoreCardMetrics
+from ..schemas import ServiceMetricCreate, ServiceMetricUpdate , ServiceMetricCL
 from .base import CRUDBase
-
+from . import CRUDScoreCardMetric, CRUDMetric
+from app.utils.utility_functions import main_calculate_error, calculate_score
+from typing import Optional
+from datetime import datetime
 
 class CRUDServiceMetric(CRUDBase[ServiceMetric, ServiceMetricCreate, ServiceMetricUpdate]):
     def __init__(self, db_session: Session):
         super(CRUDServiceMetric, self).__init__(ServiceMetric, db_session)
+        self.scorecardMetrics = CRUDScoreCardMetric(db_session)
+        self.Metric = CRUDMetric(db_session)
 
     def getByScorecardId(self, scorecardId: int) -> list[ServiceMetric]:
         return self.db_session.query(ServiceMetric).filter(ServiceMetric.scorecardId == scorecardId).all()
 
     def getByServiceId(self, serviceId: int) -> list[ServiceMetric]:
         return self.db_session.query(ServiceMetric).filter(ServiceMetric.serviceId == serviceId).all()
 
+    def get_metrics_by_scorecard_and_service(self, scorecard_id: int, service_id: int) -> list[ServiceMetric]:
```

Comment:
this function is not used. BTW, It's not correct  : )
please remove it : )

Reply:
ok